app/src/main/java/com/elkcreek/rodneytressler/intermediatepokedex/ui/PokedexView/PokedexPresenter.java

```diff
@@ -53,39 +49,48 @@ public void searchClicked(String pokemonName) {
         view.showProgressBar();
         view.clearPokemonInput();
 
-        checkDatabase(pokemonName);
+        findPokemon(pokemonName);
+    }
+
+    private Observable<PokemonApi.Pokemon> findInDatabase(String name) {
+        return pokemonDatabaseService.findPokemonByName(name).toObservable();
     }
 
-    @SuppressLint("CheckResult")
-    public void checkDatabase(String pokemonName) {
-        disposable.add(pokemonDatabaseService.findPokemonByName(pokemonName)
-                .subscribe(pokemon -> {
-                    view.hideProgressBar();
-                    view.showPokemonSprite(pokemon.getSpriteUrl());
-                    view.showPokemonName(pokemon.getPokemonName());
-                    view.showPokemonHeight("Height - " + pokemon.getPokemonHeight());
-                    view.showPokemonWeight("Weight - " + pokemon.getPokemonWeight());
-                }, throwable -> {
-                    performNetworkCall(pokemonName);
-                }));
+    private Observable<PokemonApi.Pokemon> fetchFromNetwork(String name) {
+        return pokemonService.getPokemon(name)
+                // save pokemon to database for next call
+                .doOnNext(pokemonDatabaseService::insertPokemon);
+    }
 
+    private Observable<PokemonApi.Pokemon> findInDatabaseElseFetchFromNetwork(String name) {
```

Comment:
Helper function that tries to read from db first, if that fails or is empty switch over to the network that will cache the value to the database for the next call